Dijkstra.java

```diff
@@ -0,0 +1,208 @@
+    public static void visit (Node current){
+        int rowNum = codes.indexOf(current.getCode());
+        ArrayList <Integer> row = distances.get(rowNum);
+        for(Node temp : unvisited){
+            if ((current.getDistance() + row.get(codes.indexOf(temp.getCode()))) < temp.getDistance()) {
+                temp.setDistance(current.getDistance() + row.get(codes.indexOf(temp.getCode())));
+                temp.setPrevNode(current);
+            }
+        }
+    }
+
+    // get the smallest element in the list (next node to visit)
+    public static Node getNext (ArrayList<Node> list){
+        Node temp = list.get(0);
+        for (Node elm : list){
+            if (elm.compareTo(temp) < 1){
+                temp = elm;
+            }
+        }
+        return temp;
+    }
+
+    // generate the path by tracing each node to its predecessor
+    public static ArrayList<String> getPath(ArrayList<Node> n, String st, String en){
+        Node e = null;
+        Node s = null;
+        Node temp = null;
+        ArrayList<String> list = new ArrayList<>();
+        for (Node elm : n){
+            if (elm.getCode().equals(en)){
+                temp = e = elm;
+            }
+            else if (elm.getCode().equals(st)){
+                s = elm;
+            }
+        }
+        while (temp != s){
+            list.add(temp.getCode());
+            temp = temp.getPrevNode();
+        }
+        list.add(s.getCode());
+        return list;
+    }
+
+
+    // a class to represent the vertices in a graph
+    public static class Node implements Comparable<Node>{
+        private String code;
+        private Integer distance;
+        private Node prevNode;
+
+        public Node(){
+            code = "";
+            distance = Integer.MAX_VALUE;
+            prevNode = null;
+        }
+
+        public Node(String c){
+            code = c;
+            distance = Integer.MAX_VALUE;
+            prevNode = null;
+        }
+
+        public Node (String c, Integer d){
+            code = c;
+            distance = d;
+            prevNode = null;
+        }
+
+        public void setDistance(Integer d){
+            distance = d;
+        }
+
+        public void setPrevNode(Node n){
+            prevNode = n;
+        }
+
+        public void setCode(String c){
+            code = c;
+        }
+
+        public Integer getDistance(){
+            return distance;
+        }
+
+        public String getCode(){
+            return code;
+        }
+
+        public Node getPrevNode(){
+            return prevNode;
+        }
+
+        @Override
+        public String toString(){
+            return (code + " " + distance + " " + prevNode);
+        }
+
+        @Override
+        public int compareTo(Node other) {
+            if (this.getDistance().equals(other.getDistance())){
+                return this.getCode().compareTo(other.getCode());
+            }
+            else {
```

Comment:
The standard is usually to put the else on the same line as the `}`. So it would look like `} else {`